cmd/kubeadm/app/cmd/completion_test.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"bytes"
+	"github.com/spf13/cobra"
+	"testing"
+)
+
+func TestNewCmdCompletion(t *testing.T) {
+	var out bytes.Buffer
+	shells := GetSupportedShells()
+	if len(shells) == 0 {
+		t.Errorf("GetSupportedShells() returned zero")
+	}
+	// test NewCmdCompletion with a valid shell.
+	// use a dummy parent command as NewCmdCompletion needs it.
+	parentCmd := &cobra.Command{}
```

Comment:
No need to have such a dummy parent cmd.

Just running `cmd.Run(cmd, []string{shells[0]})`.

Reply:
i think the above will `SIGSEGV` when a call like this is reached `kubeadm.GenBashCompletion()` in `completion.go`.
the command that reaches `completion.go` needs a parent, so that is the purpose of the dummy parent command. why `.Parent()` is used in `completion.go` - i have no idea.